### PR TITLE
Apple auto-connect

### PR DESCRIPTION
**Pull request by alalamav**
* Enables the auto-connect feature for iOS and macOS. 
  * Configures the VPN to connect on demand to any available network, allowing the VPN to automatically connect on boot if it was connected at shutdown.
  * Enables the macOS launcher application to start the app's UI.
  * The VPN can no longer be managed through Settings/System Preferences.
* Bypasses the connectivity checks for auto-connect.

### CODE REVIEW

File: cordova-plugin-outline/apple/src/OutlineVpn.swift [hunk 1]
```diff
@@ -165,9 +166,12 @@ class OutlineVpn: NSObject {
         self.onStartVpnExtensionMessage(response, completion: completion)
       }
       var config: [String: String]? = nil
-      if connectionId != nil {
+      if !isAutoConnect {
         config = connection.config
         config?[MessageKey.connectionId] = connectionId
+      } else {
+        // macOS app was started by launcher.
+        config = ["is-on-demand": "true"];
```
Comment (bemasc) [inline cordova-plugin-outline/apple/src/OutlineVpn.swift]: This constant is still here.  Could it be something like `MessageKey.isOnDemand`?

File: cordova-plugin-outline/apple/vpn/Shadowsocks.m [hunk 2]
```diff
@@ -61,6 +61,7 @@ @interface Shadowsocks () <GCDAsyncSocketDelegate, GCDAsyncUdpSocketDelegate>
 @property (nonatomic) bool serverCredentialsAreValid;
 @property (nonatomic) bool isServerReachable;
 @property (nonatomic) int udpForwardingNumChecks;
+@property(nonatomic) bool performConnectivityChecks;
```
Comment (bemasc) [inline cordova-plugin-outline/apple/vpn/Shadowsocks.m]: Nit: space

File: cordova-plugin-outline/apple/src/OutlineVpn.swift (hunk 1 above)
Comment (alalamav) [reply cordova-plugin-outline/apple/src/OutlineVpn.swift]: Good catch, done.

File: cordova-plugin-outline/apple/vpn/Shadowsocks.m (hunk 2 above)
Comment (alalamav) [reply cordova-plugin-outline/apple/vpn/Shadowsocks.m]: Done. clang-format should have caught it :/

**Pull request merged**